bridge/github/export.go

```diff
@@ -0,0 +1,731 @@
+package github
+
+import (
+	"bytes"
+	"context"
+	"encoding/json"
+	"fmt"
+	"io/ioutil"
+	"net/http"
+	"time"
+
+	"github.com/pkg/errors"
+	"github.com/shurcooL/githubv4"
+
+	"github.com/MichaelMure/git-bug/bridge/core"
+	"github.com/MichaelMure/git-bug/bug"
+	"github.com/MichaelMure/git-bug/cache"
+	"github.com/MichaelMure/git-bug/util/git"
+)
+
+var (
+	ErrMissingIdentityToken = errors.New("missing identity token")
+)
+
+// githubExporter implement the Exporter interface
+type githubExporter struct {
+	conf core.Configuration
+
+	// number of exported bugs
+	exportedBugs int
+
+	// number of exported labels
+	exportedLabels int
+
+	// export only bugs taged with one of these origins
+	onlyOrigins []string
+
+	// cache identities clients
+	identityClient map[string]*githubv4.Client
+
+	// map identities with their tokens
+	identityToken map[string]string
+
+	// github repository ID
+	repositoryID string
+
+	// cache identifiers used to speed up exporting operations
+	// cleared for each bug
+	cachedIDs map[string]string
+
+	// cache labels used to speed up exporting labels events
+	cachedLabels map[string]string
+}
+
+// Init .
+func (ge *githubExporter) Init(conf core.Configuration) error {
+	ge.conf = conf
+	//TODO: initialize with multiple tokens
+	ge.identityToken = make(map[string]string)
+	ge.identityClient = make(map[string]*githubv4.Client)
+	ge.cachedIDs = make(map[string]string)
+	ge.cachedLabels = make(map[string]string)
+	return nil
+}
+
+// allowOrigin verify that origin is allowed to get exported.
+// if the exporter was initialized with no specified origins, it will return
+// true for all origins
+func (ge *githubExporter) allowOrigin(origin string) bool {
+	if ge.onlyOrigins == nil {
```

Comment:
```suggestion
	if len(ge.onlyOrigins) == 0 {
```
an empty array is not nil